config-provisioning/src/main/java/com/yahoo/config/provision/Provisioner.java

```diff
@@ -25,6 +25,13 @@
      */
     List<HostSpec> prepare(ApplicationId applicationId, ClusterSpec cluster, Capacity capacity, int groups, ProvisionLogger logger);
 
+    /**
+     * fetch or create a port allocator for the given host
+     * @param host the host spec
+     * @return the port allocator
+     **/
+    default NetworkPorts getNetworkPorts(HostSpec host) { return new NetworkPorts(); }
```

Comment:
I don't think this should be a separate method on this level, but part of the payload inside HostSpec such that existing ports are returned as part of the hosts in prepare, and the wanted complete list of ports is passed back to the node repo in activate.